docs/concepts/feathr-concepts-for-beginners.md

```diff
@@ -0,0 +1,115 @@
+
+In this case, how would we "link" the `observation` dataset, and the "additional dataset"? This is called [Feature Join](#why-does-feathr-need-feature-join-and-feature-query) in Feathr, but basically think this process as joining two tables.
+
+Since this is a table join process, we need to specify which `key(s)` that the join would happen. Those `keys` are usually some IDs, but can be others as well. In the above example, if we want to augment the user click stream data with user purchase history, we will use the user ID as `key`, so that the `click_stream` table and the `historical_buying` table can be joined together. Another case might be we want to get the historical blood pressure history for a patient, so `patient ID` will be used as key in that case. That's why you need to specify `keys` in Feathr `Feature`, becasue you will need to join your `Feature` with your `Observation Data` later on.
+
+Since those additional features are from different sources, we want to define an `Anchor` to process it further. Think `Anchor` as a `Feature View`, where it is a collection of features and their corresponding sources. Think `Feature` just as a column in your dataset but it contains some useful information that you want to use in your machine learning scenario.
+
+`Source` in Feathr just represents the source data that you will need to use to extract features from. It also comes with handy customizations that you can run almost arbitrary PySpark/SparkSQL code.
+
+That's why you will see something like below, where you define a Feathr Source (in this case it's an HDFS like source) and corresponding features, and then compose an `Anchor` that combines the `Features` and `Source`.
+
+```python
+batch_source = HdfsSource(name="nycTaxiBatchSourcePurview",
+                          path="wasbs://public@azurefeathrstorage.blob.core.windows.net/sample_data/feathr_delta_table",
+                          event_timestamp_column="lpep_dropoff_datetime",
+                          timestamp_format="yyyy-MM-dd HH:mm:ss")
+features = [
+    Feature(name="f_is_long_trip_distance",
+            feature_type=BOOLEAN,
+            transform="cast_float(trip_distance)>30"),
+    Feature(name="f_day_of_week",
+            feature_type=INT32,
+            transform="dayofweek(lpep_dropoff_datetime)"),
+]
+
+request_anchor = FeatureAnchor(name="request_features",
+                               source=batch_source,
+                               features=features)
+```
+
+## Motivation on `Derived Feature`
+
+That sounds all good, but what if we want to share a feature, and others want to build additional features on top of that feature? Thats's why there is a concept in Feathr called `derived feature`, which allows you to calculate features based on other features, with certain transformation support. In practice, people can build features on top of other features and have a "feature chain".
```

Comment:
i feel we don't need to mention sharing here.

Maybe give two more examples of derived features. Like embedding multiplication. That will be easier to tell the story.